### PR TITLE
Revert "Feat/check allownace on create"

### PR DESCRIPTION
**Pull request by Julio-SDSharp**

### CODE REVIEW

File: frontend/@types/allowance.ts [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
+    name: z.string().optional(),
+  }),
   expiration: z.string().optional(),
+  noExpire: z.boolean(),
 });
 
 export type TAllowance = z.infer<typeof allowanceSchema>;
 
-export const AllowancesTableColumnsEnum = z.enum(["subAccountId", "spender", "amount", "expiration", "action"]);
+export const AllowancesTableColumnsEnum = z.enum(["subAccount", "spender", "amount", "expiration", "action"]);
```
Comment (OliverSDSharp) [inline frontend/@types/allowance.ts]: Why Change it to subAccount? is not an standard to use subAccountId?

File: frontend/components/input/CurrencyInput.tsx [hunk 2]
```diff
@@ -24,7 +26,7 @@ export default function CurrencyInput(props: InputCurrencyProps) {
         className="w-4/5 outline-none bg-inherit"
         autoComplete="false"
         onChange={handleCurrencyChange}
-        value={value || ""}
+        value={amount || value || ""}
```
Comment (OliverSDSharp) [inline frontend/components/input/CurrencyInput.tsx]: value gets updated with onCurrencyChange function?  because if does, here we have a doble update and could cause rendering issues.

File: frontend/pages/home/components/ICRC/allowance/UpdateForm/AmountFormItem.tsx [hunk 3]
```diff
@@ -31,6 +29,10 @@ export default function AmountFormItem(props: IAmountFormItemProps) {
     };
   }, [allowance, selectedAsset]);
 
+  const onAmountChange = (amount: string) => {
```
Comment (OliverSDSharp) [inline frontend/pages/home/components/ICRC/allowance/UpdateForm/AmountFormItem.tsx]: function instead of const please

File: frontend/pages/home/components/ICRC/allowance/UpdateForm/ExpirationFormItem.tsx [hunk 4]
```diff
@@ -14,7 +13,17 @@ interface IExpirationFormItemProps {
...
   const { t } = useTranslation();
   const { setAllowanceState, isLoading, allowance } = props;
-  const hasExpiration = useMemo(() => Boolean(allowance?.expiration), [allowance]);
+
+  const onDateChange = (date: dayjs.Dayjs | null) => {
+    if (!date) return;
+    setAllowanceState({ ...allowance, expiration: date.format() });
+  };
+
+  const onExpirationChange = (checked: boolean) => {
```
Comment (OliverSDSharp) [inline frontend/pages/home/components/ICRC/allowance/UpdateForm/ExpirationFormItem.tsx]: why the change to const functions?

**Review by OliverSDSharp — CHANGES_REQUESTED** [review]

File: frontend/@types/allowance.ts (hunk 1 above)
Comment (JulioSDS) [reply frontend/@types/allowance.ts:30]: This is the name of a table column, the column show the sub account id and the sub account name. It should be subAccount instead of subAccountId.
